universal-application-tool-0.0.1/app/views/admin/questions/QuestionsListView.java

```diff
@@ -1,100 +1,120 @@
-        body()
-            .with(div(maybeFlash.orElse("")))
-            .with(renderHeader("All Questions"))
-            .with(questionContent)
-            .with(renderAddQuestionLink()));
-  }
-
-  /** Renders a page with either a table view of all questions. */
-  public Content renderAsTable(
-      ImmutableList<QuestionDefinition> questions, Optional<String> maybeFlash) {
-    return render(
-        ImmutableList.of(renderQuestionTable(questions), renderSummary(questions)), maybeFlash);
-  }
-
-  /** Renders a page with either a list view of all questions. */
-  public Content renderAsList(
-      ImmutableList<QuestionDefinition> questions, Optional<String> maybeFlash) {
-    ImmutableList.Builder<Tag> builder = ImmutableList.builder();
-    for (QuestionDefinition qd : questions) {
-      builder.add(renderQuestionDefinitionInfo(qd));
-    }
-    return render(builder.build(), maybeFlash);
+        div(maybeFlash.orElse("")),
+        renderHeader("All Questions"),
+        div(renderQuestionTable(questions)).withClasses(Styles.M_4),
+        renderAddQuestionLink(),
+        renderSummary(questions));
   }
 
   private Tag renderAddQuestionLink() {
-    return a("Create a new question").withHref("/admin/questions/new");
+    return renderLinkButton("Create new question", "/admin/questions/new").withId("createQuestion");
   }
 
   private Tag renderSummary(ImmutableList<QuestionDefinition> questions) {
-    return div("Total Questions: " + questions.size());
-  }
-
-  /** Renders a div with some basic QuestionDefinition info in it. */
-  private Tag renderQuestionDefinitionInfo(QuestionDefinition definition) {
-    return div()
-        .with(div(definition.getName()))
-        .with(div(definition.getDescription()))
-        .with(div(definition.getQuestionType().toString()));
+    return div("Total Questions: " + questions.size())
+        .withClasses(Styles.FLOAT_RIGHT, Styles.TEXT_BASE, Styles.PX_4, Styles.MY_2);
   }
 
   /** Renders the full table. */
   private Tag renderQuestionTable(ImmutableList<QuestionDefinition> questions) {
     return table()
+        .withClasses(Styles.BORDER, Styles.BORDER_GRAY_300, Styles.SHADOW_MD, Styles.W_FULL)
         .with(renderQuestionTableHeaderRow())
         .with(tbody(each(questions, question -> renderQuestionTableRow(question))));
   }
 
   /** Render the question table header row. */
   private Tag renderQuestionTableHeaderRow() {
-    return thead(each(tableCells, tableCell -> th(tableCell.getHeaderText())));
+    return thead(
+        tr().withClasses(Styles.BORDER_B, Styles.BG_GRAY_200, Styles.TEXT_LEFT)
+            .with(th("Info").withClasses(BaseStyles.TABLE_CELL_STYLES, "w-2/5"))
+            .with(th("Question text").withClasses(BaseStyles.TABLE_CELL_STYLES, "w-2/5"))
+            .with(
+                th("Actions")
+                    .withClasses(
+                        BaseStyles.TABLE_CELL_STYLES, Styles.TEXT_RIGHT, Styles.PR_8, "w-1/5")));
```

Comment:
Suggestion (blocking): Here and L65 if a tailwind class is missing from `Styles` please add it and reference rather than using a raw string.

Reply:
Done.